### ISSUE TITLE
Dash stream doesn't play after playing mp4

### ISSUE THREAD

**Issue by bartoszwilk** [brianwernick/ExoMedia#579]
- [+] I have verified there are no duplicate active or recent bugs, questions, or requests

###### Include the following:
 - ExoMedia version: `4.0.1-4.1.0 (4.0.0 works fine)`
 - Device OS version: `API 17+`
 - Device: `Any`
 
###### Reproduction Steps
1. Download and run this [showcase project](https://github.com/bartoszwilk/ExoMediaDashShowcase)
2. Click "Dash after MP4" button
3. Notice that dash stream won't be played after mp4 clips

###### Expected Result
Dash stream is playing after mp4 clips

###### Actual Result
VideoView is entering into idle state right after initializing dash stream

I added additional scenarios so you can see that everything works fine when dash and mp4 clips are playing separately.
Dash stream works correctly when playing standalone. As well as mp4 clips do.
ListenerMux state changes (ListenerMux.onStateChanged) below:
Scenario with 2 mp4s and dash stream after them
1. Playing first mp4
STATE_BUFFERING
STATE_READY
(video is playing...)
STATE_ENDED
2. Playing second mp4
STATE_BUFFERING
STATE_READY
(video is playing...)
STATE_ENDED
3. Starting dash stream playing
STATE_BUFFERING
STATE_ENDED
STATE_IDLE

No error is being emitted also.

EDIT: I also added "hls after mp4" scenario which works fine

**Comment by brianwernick** [comment]
Hmm, I added the urls to the demo app samples and didn't see the issue with my Nexus 6P (8.1.0) with the current `master` branch.
```java
        videoSamples.add(new Sample("579-1", "http://www.sample-videos.com/video/mp4/720/big_buck_bunny_720p_2mb.mp4"));
        videoSamples.add(new Sample("579-2", "http://techslides.com/demos/sample-videos/small.mp4"));
        videoSamples.add(new Sample("579-3", "http://rdmedia.bbc.co.uk/dash/ondemand/elephants_dream/1/client_manifest-all.mpd"));
```

I'll look in to it more (try released versions, your demo, more devices, etc.) when I get some time; it might be that the updated version of ExoPlayer in `master` has fixed the issue

**Comment by bartoszwilk** [comment]
Everything works fine after I've added exomedia as a module to my showcase project.
So probably you we right, the problem could be related with exoplayer.
I'm waiting for the next release then :) Thanks for the help.

**Comment by brianwernick** [comment]
Ok. I'll mark it as resolved then for the 4.2.0 release
